src/metabase/api/search.clj

```diff
@@ -409,32 +357,39 @@
   [instance]
   (mi/can-read? instance))
 
-(mu/defn query-model-set
+(mu/defn query-model-set :- [:set SearchableModel]
   "Queries all models with respect to query for one result to see if we get a result or not"
   [search-ctx :- SearchContext]
-  (map #(get (first %) :model)
-       (filter not-empty
-               (for [model search-config/all-models]
-                 (let [search-query     (search-query-for-model model search-ctx)
-                       query-with-limit (sql.helpers/limit search-query 1)]
-                   (mdb.query/query query-with-limit))))))
+  (let [model-queries (for [model (:models search-ctx)]
```

Comment:
This will improve the performance a bit. Previously it did multiple roundtrips to the DB, now it only does one using a nested union query.